### ISSUE TITLE
Missing var from "for (var key in foo)"

### ISSUE THREAD

**Issue by SystemParadox** [babel/babel#1819]
Given:
```
try {
    for (var key in foo) {
    }
    let defaultAdapter;
    var baz = function() {
        let adapter = defaultAdapter;
    }
} catch (err) {
}
```

Babel will output the following:

```
"use strict";

try {
    var baz;

    (function () {
        for (key in foo) {}
        var defaultAdapter = undefined;

        baz = function baz() {
            var adapter = defaultAdapter;
        };
    })();
} catch (err) {}
```

Notice that the var declaration is missing for `key` in the for loop. This results in a global leak. It's ok using `let key` instead.

I've played around with this a lot and cannot reduce the example any further.

Thanks.

**Comment by sebmck** [comment]
Fixed as of 5.6.7, thanks!

**Comment by SystemParadox** [comment]
Excellent. Thanks very much.
